packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -64,35 +107,21 @@ const AccountRecoveryResetPassword = ({
       },
     });
 
-  const alertSuccessAndNavigate = useCallback(() => {
-    const successCompletePwdReset = ftlMsgResolver.getMsg(
-      `${viewName}-success-alert`,
-      'Password set'
+  // Make some presentation decisions based on initial state
+  if (state.contextError) {
+    alertInvalidContext(state.contextError);
+  } else if (!state.supportsRecovery) {
+    const msg = ftlMsgResolver.getMsg(
+      'redirecting',
+      'Redirecting'
     );
-    alertBar.success(successCompletePwdReset);
-    navigate('/reset_password_with_recovery_key_verified', { replace: true });
-  }, [alertBar, ftlMsgResolver, navigate]);
-
-  const sendNewLinkEmail = () => {
-    // TODO: Hook up functionality to send a new verification link
-  };
-
-  const onSubmit = () => {
-    try {
-      // TODO: - completeAccountPasswordResetWithRecoveryKey
-      //       - logViewEvent('verification.success');
-      //       - metrics.logUserPreferences('account-recovery', false)
-      //       - logFlowEventOnce('recovery-key-consume.success', viewName);
-      alertSuccessAndNavigate();
-    } catch (e) {
-      // TODO: - if token expired, re-render and show LinkExpired
-      //       - metrics event for error
-      //       - error feedback in alertBar
-    }
-  };
+    alertBar.info(msg);
+    navigate(`/complete_reset_password?${urlSearchContext.toSearchQuery()}`);
```

Comment:
@millsoper this passes query parameter state between 'views', since the models like verificationInfo and reliers are using urlSearchContext to hold their current state.